manifests/plugin/versionlock.pp

```diff
@@ -0,0 +1,32 @@
+# Class: yum::plugin::versionlock
+#
+# This class sets the structure for the lock file
+#
+# Parameters:
+#   [*ensure*] - specifies if versionlock should be present or absent
+#   [*clean*] - specifies if yum clean all should be called after edits. Defaults false.
+#
+# Actions:
+#
+# Requires:
+#
+# Sample usage:
+#   include zypprepo::plugin::versionlock
+#
+class zypprepo::plugin::versionlock (
+  Enum['present', 'absent'] $ensure = 'present',
+  String                    $path   = '/etc/zypp/locks',
```

Comment:
Any reason not to use `Stdlib::Absolutepath` data type ?

Reply:
This was copied from the Yum module. As we are importing the stdlib library, this can indeed be the AbsolutePath datatype. I will make this change.